testresults/get.go

```diff
@@ -46,19 +50,32 @@ func GetTestResults(ctx context.Context, opts TestResultsGetOptions) ([]byte, er
 	}
 
 	var urlString string
-	if opts.DisplayTaskID != "" {
+	if opts.DisplayTaskID != "" && !opts.FailedSample {
 		urlString = fmt.Sprintf("%s/rest/v1/test_results/display_task_id/%s", opts.CedarOpts.BaseURL, url.PathEscape(opts.DisplayTaskID))
 	} else if opts.TestName == "" {
 		urlString = fmt.Sprintf("%s/rest/v1/test_results/task_id/%s", opts.CedarOpts.BaseURL, url.PathEscape(opts.TaskID))
 	} else {
 		urlString = fmt.Sprintf("%s/rest/v1/test_results/test_name/%s/%s", opts.CedarOpts.BaseURL, url.PathEscape(opts.TaskID), url.PathEscape(opts.TestName))
 	}
-	urlString += fmt.Sprintf("?execution=%d", opts.Execution)
+	if opts.FailedSample {
+		urlString += "/failed_sample"
+	}
+
+	var params string
+	if !opts.LatestExecution {
+		params += fmt.Sprintf("execution=%d", opts.Execution)
+	}
+	if opts.FailedSample && opts.DisplayTaskID != "" {
+		params += "display_task=true"
+	}
+	if len(params) > 0 {
+		urlString += "?" + params
+	}
```

Comment:
The URL-building logic has become sufficiently complicated over multiple changes that it might be worth unit testing on its own to ensrue that URLs are built correctly for each options combination. I had some difficulty reasoning through whether every single case produced a valid URL or not.

Reply:
Yeah I agree. I am going to hold off on writing tests until I simplify the test results URL in https://jira.mongodb.org/browse/EVG-15299.